scripts/zephyr_module.py

```diff
@@ -245,35 +246,106 @@ def process_twister(module, meta):
     return out
 
 
+def process_meta(zephyr_base, west_projects, modules):
+    # Process zephyr_base, projects, and modules and create a dictionary
+    # with meta information for each input.
+    #
+    # The dictionary will contain meta info in the following lists:
+    # - zephyr:        path and revision
+    # - modules:       name, path, and revision
+    # - west-projects: path and revision
+    #
+    # returns the dictionary with said lists
+
+    meta = {'zephyr': None, 'modules': None, 'west': None}
+
+    def git_revision(path):
+        rc = subprocess.Popen(['git', 'rev-parse', '--is-inside-work-tree'],
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE,
+                              cwd=path).wait()
+        if rc == 0:
+            # A git repo.
+            popen = subprocess.Popen(['git', 'rev-parse', 'HEAD'],
+                                     stdout=subprocess.PIPE,
+                                     stderr=subprocess.PIPE,
+                                     cwd=path)
+            stdout, stderr = popen.communicate()
+            stdout = stdout.decode('utf-8')
+
+            if not (popen.returncode or stderr):
+                revision = stdout.rstrip()
+
+                rc = subprocess.Popen(['git', 'diff-index', '--quiet', 'HEAD',
+                                       '--'],
+                                      stdout=None,
+                                      stderr=None,
+                                      cwd=path).wait()
+                if rc:
+                    return revision + '-dirty'
+                return revision
```

Comment:
yes, but I see two issues with such approach:
1. Not all repos are tagged, resulting in `fatal: No names found, cannot describe anything.` when no tag can be found.
    This means that some repos would have full revision, others a `describe` ==> inconsistency in the file.
2. A tag may be a local only tag, or when same workspace is used for different workspace then same SHA could have multiple tags, but the tag presented with git describe could be wrong, for example:
   ```
    $ git checkout v2.6.0
    HEAD is now at 79a6c07536 release: Updates for v2.6.0
    $ git describe --abbrev=12
    zephyr-v2.6.0   # <--- Notice, this was not the tag used when I did the checkout.
    ```
    which happens because NCS has tagged the same commit with `zephyr-v2.6.0` as Zephyr tagged with `v2.6.0`.
    So if i'm working in vanilla Zephyr, and suddenly have a `NCS` Zephyr tag in the build, that would be misleading and could cause confusion, so the SHA is safer. And the SHA is valid both in vanilla Zephyr and NCS Zephyr.